pkg/controller/podgc/gc_controller.go

```diff
@@ -335,7 +331,9 @@ func (gcc *PodGCController) markFailedAndDeletePod(ctx context.Context, pod *v1.
 
 func (gcc *PodGCController) markFailedAndDeletePodWithCondition(ctx context.Context, pod *v1.Pod, condition *corev1apply.PodConditionApplyConfiguration) error {
 	klog.InfoS("PodGC is force deleting Pod", "pod", klog.KRef(pod.Namespace, pod.Name))
-	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+	// See https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated#risks-and-mitigations
```

Comment:
Give a little bit of context.  Maybe something like:

Patch the pod to make sure it is transitioned to the `Failed` phase before deletion. This is needed for the `JobPodReplacementPolicy` feature to make sure Job replacement pods are created. See ... for more details.

Reply:
done.